plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2370,71 +2302,21 @@ bool producer_plugin_impl::process_unapplied_trxs(const fc::time_point& deadline
+   int   num_applied    = 0;
+   int   num_failed     = 0;
+   int   num_processed  = 0;
+   bool  exhausted      = false;
 
-   auto&              blacklist_by_id     = _blacklisted_transactions.get<by_id>();
    chain::controller& chain               = chain_plug->chain();
    time_point         pending_block_time  = chain.pending_block_time();
-   auto               end                 = _unapplied_transactions.incoming_end();
-   const auto&        sch_idx             = chain.db().get_index<generated_transaction_multi_index, by_delay>();
-   const auto         scheduled_trxs_size = sch_idx.size();
-   auto               sch_itr             = sch_idx.begin();
-   while (sch_itr != sch_idx.end()) {
-      if (sch_itr->delay_until > pending_block_time)
-         break; // not scheduled yet
-      if (exhausted || deadline <= fc::time_point::now()) {
-         exhausted = true;
-         break;
-      }
-      if (sch_itr->published >= pending_block_time) {
-         ++sch_itr;
-         continue; // do not allow schedule and execute in same block
-      }
+   const auto&        expired_idx         = chain.db().get_index<generated_transaction_multi_index, by_expiration>();
+   const auto expired_size                = expired_idx.size();
+   auto               expired_itr         = expired_idx.begin();
 
-      if (blacklist_by_id.find(sch_itr->trx_id) != blacklist_by_id.end()) {
-         ++sch_itr;
-         continue;
-      }
-
-      const transaction_id_type trx_id         = sch_itr->trx_id; // make copy since reference could be invalidated
-      const auto                sch_expiration = sch_itr->expiration;
-      auto                      sch_itr_next   = sch_itr; // save off next since sch_itr may be invalidated by loop
-      ++sch_itr_next;
-      const auto next_delay_until = sch_itr_next != sch_idx.end() ? sch_itr_next->delay_until : sch_itr->delay_until;
-      const auto next_id          = sch_itr_next != sch_idx.end() ? sch_itr_next->id : sch_itr->id;
-
-      num_processed++;
-
-      // configurable ratio of incoming txns vs deferred txns
-      while (incoming_trx_weight >= 1.0 && itr != end) {
-         if (deadline <= fc::time_point::now()) {
-            exhausted = true;
-            break;
-         }
-
-         incoming_trx_weight -= 1.0;
-
-         auto trx_meta = itr->trx_meta;
-         bool api_trx  = itr->trx_type == trx_enum_type::incoming_api;
-
-         auto trx_tracker = _time_tracker.start_trx(trx_meta->is_transient());
-         push_result pr = push_transaction(deadline, trx_meta, api_trx, itr->return_failure_trace, trx_tracker, itr->next);
-
-         exhausted = pr.block_exhausted;
-         if (pr.trx_exhausted) {
-            ++itr; // leave in incoming
-         } else {
-            itr = _unapplied_transactions.erase(itr);
-         }
-
-         if (exhausted)
-            break;
+   while (expired_itr != expired_idx.end()) {
+      if (expired_itr->expiration > pending_block_time) {
```

Comment:
Needs to be `>=`

Reply:
See `controller::validate_expiration` which has `EOS_ASSERT( trx.expiration.to_time_point() >= pending_block_time()`

Reply:
Thanks!